devito/operator.py

```diff
@@ -186,6 +186,22 @@ def signature(self):
         return self.input_params + [param for param in self.output_params
                                     if param not in self.input_params]
 
+    @property
+    def total_kernel_time(self):
+        return self.propagator.timings[0]
+
+    @property
+    def total_space_loops_time(self):
+        return self.propagator.timings[1]
+
+    @property
+    def total_pre_loops_time(self):
+        return self.propagator.timings[2]
+
+    @property
+    def total_post_loops_time(self):
```

Comment:
pre_loop and post_loop are operations done outside of the spacial loop :
for (t=...)
    pre_loop
    for (x=.., y=...)
        stencils, factors,...
    endforxyz
    post_loop
endfort